src/pybind/mgr/diskprediction/agent/metrics/sai_disk.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import absolute_import
+
+import socket
+
+from . import MetricsAgent, AGENT_VERSION
+from ...common.db import DB_API
+from ...models.metrics.dp import SAI_Disk
+
+
+def get_human_readable(size, precision=2):
+    suffixes = ['B', 'KB', 'MB', 'GB', 'TB']
+    suffix_index = 0
+    while size > 1024 and suffix_index < 4:
+        # increment the index of the suffix
+        suffix_index += 1
+        # apply the division
+        size = size/1000.0
+    return "%.*d %s" % (precision, size, suffixes[suffix_index])
+
+
+class SAI_DiskAgent(MetricsAgent):
+    measurement = 'sai_disk'
+
+    @staticmethod
+    def _get_disk_type(is_ssd, vendor, model):
+        """ return type:
+            0: "Unknown", 1: "HDD",
+            2: "SSD",     3: "SSD NVME",
+            4: "SSD SAS", 5: "SSD SATA",
+            6: "HDD SAS", 7: "HDD SATA"
+        """
+        if is_ssd:
+            if vendor:
+                disk_type = 4
+            elif model:
+                disk_type = 5
+            else:
+                disk_type = 2
+        else:
+            if vendor:
+                disk_type = 6
+            elif model:
+                disk_type = 7
+            else:
+                disk_type = 1
+        return disk_type

```

Comment:
this never returns 3?

Reply:
Reference pysmart module, it did not handle the nvme device.

Reply:
smartctl did not support nvme, we define this logic not include nvme.
 
[root@localhost ~]# smartctl -A /dev/nvme0
smartctl 6.6 2017-11-05 r4594 [x86_64-linux-3.10.0-514.el7.x86_64] (local build)

Copyright (C) 2002-17, Bruce Allen, Christian Franke, www.smartmontools.org
  
=== START OF SMART DATA SECTION ===
Read NVMe SMART/Health Information failed: NVMe Status 0x04

Reply:
@liewegas I refine this at latest changed. It should better.